DIP/DIP/Models/CardPaymentService.cs

```diff
@@ -0,0 +1,22 @@
+﻿using DIP.Interfaces;
+
+namespace DIP
+{
+    public class CardPaymentService<CardPayment> : IPaymentService<CardPayment> where CardPayment : IPayment
+    {
+        public string Pay(CardPayment payment )
+        {
+            if (ValidDiscount(payment.Discount)) {
+                double totalAfterDiscount = payment.Total - (payment.Total * payment.Total / 100);
```

Comment:
a more readable way to write this code would be something similar to this:
var discountAmount = payment.Total * payment.Total / 100;
double totalAfterDiscount = payment.Total - discountAmount ;

this way you can easily understand what the code should do, without looking at the formula.

also, amount should be decimal.